packages/data-stores/src/plans/queries/lib/unpack-intro-offer.ts

```diff
@@ -0,0 +1,40 @@
+import type { PlanIntroductoryOffer, PricedAPIPlan, PricedAPISitePlan } from '../../types';
+
+/**
+ * Unpacks an intro offer from an API plan to the respective `PlanIntroductoryOffer` structure for the store.
+ */
+const unpackIntroOffer = (
+	plan: PricedAPISitePlan | PricedAPIPlan
+): PlanIntroductoryOffer | null => {
+	// these aren't grouped or separated. so no introductory offer if no cost or interval
+	if ( ! plan.introductory_offer_interval_unit && ! plan.introductory_offer_interval_count ) {
+		return null;
+	}
+
+	/**
+	 * An offer is complete if:
+	 *   1. an expiry date is set on the plan that exceeds the offer end date, OR
+	 *   2. the current Date exceeds the offer end date
+	 */
+	const isOfferComplete =
+		'expiry' in plan
+			? Boolean(
+					plan.expiry &&
+						plan.introductory_offer_end_date &&
+						new Date( plan.expiry ) > new Date( plan.introductory_offer_end_date )
+			  )
+			: Boolean(
+					plan.introductory_offer_end_date &&
+						new Date() > new Date( plan.introductory_offer_end_date )
+			  );
```

Comment:
I am not sure if there is any need to keep a complete or expired offer in the pricing data structure. My guess is we could return null here and get rid of the `isOfferComplete` property. cc @daledupreez (since you introduced this)

Reply:
There is way more context in https://github.com/Automattic/wp-calypso/pull/82517 where I added these properties, but the reason is mostly to have the plans grid correctly show the plan price based on whether you're still within the introductory offer period and pricing, or whether you've already renewed at the full price.

Reply:
Thanks @daledupreez. I will review the test cases you added in https://github.com/Automattic/wp-calypso/pull/82517. It might merit a refactor if we can just not attach the intro offer if expired (vs mark it as `isOfferComplete`) - if a "complete offer" is not rendered anywhere 🤔